tpm2/credactivation/credential_activation.go

```diff
@@ -137,3 +135,63 @@ func generateRSA(aik *tpm2.HashValue, pub *rsa.PublicKey, symBlockSize int, secr
 
 	return packedID, packedEncSecret, nil
 }
+
+func createRSASeed(aik *tpm2.HashValue, ek *rsa.PublicKey, symBlockSize int, rnd io.Reader) ([]byte, []byte, error) {
+	crypothash, err := aik.Alg.Hash()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// The seed length should match the keysize used by the EKs symmetric cipher.
+	// For typical RSA EKs, this will be 128 bits (16 bytes).
+	// Spec: TCG 2.0 EK Credential Profile revision 14, section 2.1.5.1.
+	seed := make([]byte, symBlockSize)
+	if _, err := io.ReadFull(rnd, seed); err != nil {
+		return nil, nil, fmt.Errorf("generating seed: %v", err)
+	}
+
+	// Encrypt the seed value using the provided public key.
+	// See annex B, section 10.4 of the TPM specification revision 2 part 1.
+	label := append([]byte(labelIdentity), 0)
+	encryptedSeed, err := rsa.EncryptOAEP(crypothash.New(), rnd, ek, seed, label)
+	if err != nil {
+		return nil, nil, fmt.Errorf("generating encrypted seed: %v", err)
+	}
+
+	encryptedSeed, err = tpmutil.Pack(encryptedSeed)
+	return seed, encryptedSeed, err
+}
+
+func createECCSeed(ak *tpm2.HashValue, ek *ecdsa.PublicKey, rnd io.Reader) (seed, encryptedSeed []byte, err error) {
+	curve := ek.Curve
+	priv, x, y, err := elliptic.GenerateKey(curve, rnd)
+	if err != nil {
+		return nil, nil, err
+	}
+	z, _ := curve.ScalarMult(ek.X, ek.Y, priv)
```

Comment:
Hot off the presses Go has a proper ECDH library: https://pkg.go.dev/crypto/ecdh#PrivateKey.ECDH

Can we use that instead of bare ScalarMult? This might require a newer Go version but personally I prefer that. Naked ECDH was like the one safety pain point in Go's otherwise amazing standard crypto library.